File: Java/Agosto/Semana-14-al-18.java
//Ejercicio Gerardo Duckwitz
  public boolean eliminarEstudiante(Estudiate estudiante){
        PreparedStatement ps;
        Connection con = getConnection();
        String sql = "DELETE FROM estudiantes2022 WHERE idestudiante2022=?"
                try{
                    ps = con.prepareStatement(sql);
                    ps.setInt(1,estudiante.getIdEstudiante());
                    ps.execute();
                    return true;
                }catch (Exception e){
                    System.out.println("Error al eliminar el estudiante" +e.getMessage());
                }finally{
                    try{
                        con.close();
                    }catch (Exception e){
                        System.out.println("Error al cerrar la conexion");
                    }

                }
                return false;
    }


    public static void main(String[] args) {
        var salir = false;
        var consola = new Scanner((System.in));
        var estudianteDao = new EstudianteDAO()
                while (!salir){
                    try{
                        mostrarMenu();
                        salir = ejecutarOpciones(consola, estudianteDao);
                    }catch (Exception e){
                        System.out.println("Ocurrio un error");
                    }

                }
    }
    private static void mostrarMenu(){
        System.out.print("""
                ********** Sistema de Estudiantes***********
                1. Listar Estudiantes
                2. Buscar Estudiantes
                3. Agregar Estudiante
                4. Modificar Estudiante
                5. Eliminar Estudiante
                6. Salir
                Elige una opcion:
                
                
                """);
    }

//Fin ejercicio Gerardo Duckwitz
*

//Ejercicio Jose Remaggi
  public boolean eliminarEstudiante(Estudiate estudiante){
        PreparedStatement ps;
        Connection con = getConnection();
        String sql = "DELETE FROM estudiantes2022 WHERE idestudiante2022=?"
                try{
                    ps = con.prepareStatement(sql);
                    ps.setInt(1,estudiante.getIdEstudiante());
                    ps.execute();
                    return true;
                }catch (Exception e){
                    System.out.println("Error al eliminar el estudiante" +e.getMessage());
                }finally{
                    try{
                        con.close();
                    }catch (Exception e){
                        System.out.println("Error al cerrar la conexion");
                    }

                }
                return false;
    }


    public static void main(String[] args) {
        var salir = false;
        var consola = new Scanner((System.in));
        var estudianteDao = new EstudianteDAO()
                while (!salir){
                    try{
                        mostrarMenu();
                        salir = ejecutarOpciones(consola, estudianteDao);
                    }catch (Exception e){
                        System.out.println("Ocurrio un error");
                    }

                }
    }
    private static void mostrarMenu(){
        System.out.print("""
                ********** Sistema de Estudiantes***********
                1. Listar Estudiantes
                2. Buscar Estudiantes
                3. Agregar Estudiante
                4. Modificar Estudiante
                5. Eliminar Estudiante
                6. Salir
                Elige una opcion:
                
                
                """);
    }

//Fin ejercicio Jose Remaggi
//Ejercicio Anabel Alesci
//Metodo para ejecutar las opciones, va a regresar un valor booleani, ya que es el que
// puede modificar el valor de la variable salir, si es verdadero termina el ciclo while
private static boolean ejecutarOpciones(Scanner consola, EstudianteDAO estudianteDAO){
  var opcion = Integer.parseInt(consola.nextLine());
  var salir = false;
  switch(opcion){
    case 1 -> {//listar estudiantes
      System.out.println("Listado de Estudiantes");
      //no muestra la informacion, solo recupera la info y regresa una lista
        var estudiantes = estudianteDAO.listarEstudiantes();//recibe el listado
        //vamos a iterar cada objeto de tipo estudiante
        estudiantes.forEach(System.out::println);//para imprimir la lista
        }//Fin Caso 1
        case 2 -> {//Buscar estudiante por id
      System.out.println("Introduce el id_estudiante a buscar: ");
      var idEstudiante = Integer.parseInt(consola.nextLine());
      var estudiante = new Estudiante(idEstudiante);
      var encontrado = estudianteDAO.buscarEstudiantePorId(estudiante);
      if(encontrado)
        System.out.println("Estudiante encontrado: "+ estudiante);
      else
        System.out.prinln("Estudiante NO encontrado: "+ estudiante);
        }//Fin caso 2
        case 3 ->{//Agregar estudiante
        System.out.println("Agregar estudiante: ");
        System.out.println("Nombre: ");
        var nombre=console.nextLine();
        System.out.println("Apellido: ");
        var apellido=console.nextLine();
        System.out.println("Telefono: ");
        var telefono=console.nextLine();
        System.out.println("Email: ");
        var email=console.nextLine();
        //crear objeto estudiante(sin id)
        var estudiante=new Estudiante(nombre,apellido,telefono,email);
        var agregado=estudianteDAO.agregarEstudiante(estudiante);
        if(agregado)
        System.out.println("Estudiante agregado: "+estudiante);
        else
        System.out.println("Estudiante NO agregado"+estudiante);
        }//Fin Caso 3
        case 4 -> {//Modificar estudiante
      System.out.println("Modificar estudiante: ");
      //Aqui lo primero es especificar cual es el id del objeto a modificar
        System.out.println("Id Estudiante: ");
        var idEstudiante =Integer.parseInt(consola.nextLine());
        System.out.println("Nombre: ");
        var nombre = consola.nextLine();
        System.out.println("Apellido: ");
        var apellido = consola.nextLine();
        System.out.println("Telefono: ");
        var telefono = consola.nextLine();
        System.out.println("Email: ");
        var email = consola.nextLine();
        //crea el objeto estudiante a modificar
        var estudiante =
        new Estudiante(idEstudiante, nombre, apellido, telefono, email);
        var modificado = estudianteDAO.modificarEstudiante(estudiante);
        if(modificado)
        System.out.println("Estudiante modificado"+ estudiante);
        else
          System.out.println("Estudiante No modificado"+ estudiante);
    }//Fin caso 4
        case 5 -> {//Eliminar estudiante
        System.out.println("Eliminar estudiante: ");
        System.out.println("Id estudiante: ");
        var idEstudiante = Integer.parseInt(consola.nextLine());
        var estudiante = new Estudiante(idEstudiante);
        var eliminado = estudianteDAO.eliminarEstudiante(estudiante);
        if(eliminado)
          System.out.println("Estudiante eliminado: "+ estudiante);
        else
          System.out.println("Estudiante NO eliminado: "+ estudiante);
        }//Fin caso 5
        case 6 ->{//salir
        System.out.println("Hasta pronto!!!");
        salir=true;
        }//fin caso 6
        default -> System.out.println("Opción no reconocida, ingrese otra opción");
        }//Fin switch
        return salir;
        }
        }//fin clase

        }
        //Fin Ejercicio Anabel Alesci
//Ejercicio Matías Villa
package UTN;

import UTN.conexion.Conexion;
import UTN.datos.EstudianteDAO;
import UTN.dominio.Estudiante;
import org.w3c.dom.ls.LSOutput;

import java.sql.SQLOutput;
import java.util.Scanner;

public class SistemaEstudiantesApp{
    public static void main(String[] args) {
        var salir = false;
        var consola = new Scanner(System.in); //leer info de la consola
        //Se crea una instancia de la clase servicio, esto lo hacemos fuera del ciclo
        var estudianteDao = new EstudianteDAO();
        while(!salir){
            try{
            mostrarMenu();
            //Este es el método ejecutarOpciones que va a devolver el boolean
            salir = ejecutarOpciones(consola, estudianteDao);//Este arreoja una exepcion
        } catch(Exception e){
            System.out.println("Ocurrió un error al ejecutar la operación"+e.getMessage);
        }

    }//Fin while
} // Fin Main

private static void mostrarMenu(){
    System.out.println("""
            ****Sistema Estudiantes****
            1.Listar Estudiantes
            2.Buscar Estudiantes
            3.Agregar Estudiantes
            4.Modificar Estudiantes
            5.Eliminar Estudiantes
            6.Salir
            Elige una opción :
            
            """);


}

//Método para ejecutar opciones,va a regresar un valor boolean
    private static boolean ejecutarOpciones(Scanner consola, EstudianteDAO estudianteDAO) {
        var opcion = Integer.parseInt(consola.nextLine()); //recibe el número atraves del Scanner y lo convierte en un INT
        var salir = false;
        switch (opcion) {
            case 1 -> { //Listar estudiantes
                System.out.println("listado de estudiantes...");
                //no muestra la información solo recupera info y regresa una lista
                var estudiantes = estudianteDAO.listarEstudiantes();//recibe el listado
                //vamos a iterar cada objeto de tipo estudiante
                estudiantes.forEach(System.out::println);//imprimir lista
            }//Fin caso 1

            case 2 -> { //Buscar estudiantes por id
                System.out.println("introduce el id_estudiante a buscar:");
                var idEstudiante = Integer.parsInt(consola.nextLine());
                var estudiante = new Estudiantes(idEstudiante);
                var encontrado = estudianteDAO.buscarEstudiantePorId(estudiante); //recibe lo que ejecuto el usuario
                if (encontrado)
                    System.out.println("Estudiante Encontrado" + estudiante);
                else
                    System.out.println("Estudiante No encontrado" + estudiante);
            }//Fin caso 2

            case 3 -> { //Agregar Estudiante
                System.out.println("Agregar Estudiante: ");
                System.out.println("Nombre: ");
                var nombre = console.nextLine();
                System.out.println("Apellido: ");
                var apellido = console.nextline();
                System.out.println("Telefono: ");
                var telefono = consola.nextLine();
                System.out.println("Email: ");
                var email = consola.nextLine();
                //crear objeto estudiante(sin id, se agrega automaticamente)
                var estudiante = new Estudiante(nombre, apellido, telefono, email);
                var agregado = estudianteDAO.agregarEstudiante(estudiante);
                if (agregado)
                    System.out.println("Estudiante Agragado" + estudiante);


            }//fin caso 3

            case 4 -> {//Modificar Estudiante
                System.out.println("Modificar Estudiante: ");
                //Lo primero es identificar el id del objeto a modificar
                System.out.println("Id estudiante: ");
                var idEstudiante = Integer.parseInt(consola.nextLine());
                System.out.println("Nombre: ");
                var nombre = console.nextLine();
                System.out.println("Apellido: ");
                var apellido = console.nextline();
                System.out.println("Telefono: ");
                var telefono = consola.nextLine();
                System.out.println("Email: ");

                //crear objeto estudiante a modificar
                var estudiante = new Estudiante(idEstudiante, nombre, apellido, telefono, email);
                var modificado = estudianteDAO.modificarEstudiante(estudiante);
                if (modificado)
                    System.out.println("Estudiante modificado" + estudiante);
                else
                    System.out.println("Estudiante NO modificado" + estudiante);
            }//Fin caso 4

            case 5 -> {//Eliminar estudiante
                System.out.println("Eliminar estudiante:");
                System.out.println("Id estudiante: ");
                var idEstudiante = Integer.parseInt(consola.nextline());
                var estudiante = new Estudiante(idEstudiante);
                var eliminado = estudianteDAO.eliminarEstudiante(estudiante);
                if (eliminado)
                    System.out.println("Estudiante eliminado" + estudiante);
                else
                    System.out.println("Estudiante No eliminado" + estudiante);
            }//Fin caso 5

            case 6 -> {//salir
                System.out.println("Hasta pronto ");
                salir = true;
            }//Fin caso salir
            default -> System.out.println("Opcion no reconocida, ingrese otra opción");
        }//Fin switch

        return salir;

            }


        }
    



//Fin Ejercicio Matías Villa

//Ejercicio Jesús Mercado

public boolean eliminarEstudiante(Estudiate estudiante){
  PreparedStatement ps;
  Connection con = getConnection();
  String sql = "DELETE FROM estudiantes2022 WHERE idestudiante2022=?"
    try{
      
        ps = con.prepareStatement(sql);
        ps.setInt(1,estudiante.getIdEstudiante());
        ps.execute();
                return true;
        }catch (Exception e){
            System.out.println("Error al eliminar el estudiante" +e.getMessage());
        }finally{
            try{
                con.close();
            }catch (Exception e){
                System.out.println("Error al cerrar la conexion");
            }

    }
    return false;
}

public static void main(String[] args) {
      var salir = false;
      var consola = new Scanner((System.in));
      var estudianteDao = new EstudianteDAO()
            while (!salir){
                try{
                    mostrarMenu();
                    salir = ejecutarOpciones(consola, estudianteDao);
                }catch (Exception e){
                    System.out.println("Ocurrio un error");
                }

            }
}
private static void mostrarMenu(){
    System.out.print("""
            ********** Sistema de Estudiantes***********
            1. Listar Estudiantes
            2. Buscar Estudiantes
            3. Agregar Estudiante
            4. Modificar Estudiante
            5. Eliminar Estudiante
            6. Salir
            Elige una opcion:
                 
            """);
}

//Fin ejercicio Jesús Mercado

File: Python/Agosto/Semana-14-al-18.py
#Ejercicio Gerardo Duckwitz
from logger_base import log
class UsuarioDAO:
    '''
    DAO -> Data Access Object para la tabla usuario
    CRUD -> Create-read-update-delete
    '''
    _SELECT = 'SELECT * FROM usuario ORDER BY id_usuario'
    _INSERTAR = 'INSERT INTO usuario(username, password) VALUES (%s,%s)'
    _ACTUALIZAR = 'UPDATE usuario SET username=%s, password=%s WHERE id_usuario=%s'
    _ELIMINAR = 'DELETE FROM usuario WHERE id_usuario=%s'

    @classmethod
    def seleccionar(cls):
        with CursorDelPool() as cursor:
            log.debug('Seleccionando usuarios')
            cursor.execute(cls._SELECT)
            registros = cursor.fetchall()
            usuarios = []
            for registro in registros:
                usuario = Usuario(registro[0], registro[1], registro[2])
                usuarios.append(usuario)
            return usuarios
    @classmethod
    def insertar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a insertar: {usuario}')
            valores = (usuario.username, usuario.password)
            cursor.execute(cls._INSERTAR, valores)
            return cursor.rowcount

    @classmethod
    def actualizar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a actualizar {usuario}')
            valores = (usuario.username, usuario.password, usuario.id_usuario)
            cursor.execute(cls._ACTUALIZAR, valores)
            return cursor.rowcount

    @classmethod
    def eliminar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a eliminar: {usuario}')
            valores=(usuario.id_usuario,)
            cursor.execute(cls._ELIMINAR,valores)
            return cursor.rowcount

#Fin ejercicio Gerardo Duckwitz


#Ejercicios Matías Villa

from capa_datos_personas.Usuario import Usuario
from capa_datos_personas.usuario_dao import UsuarioDAO
from logger_base import log


opcion = None
while opcion != 5:
    print('Opciones: ')
    print('1.Listar Usuarios')
    print('2.Agregar Usuarios')
    print('3.Modificar usuarios')
    print('4.Eliminar usuario')
    print('5.Salir')
    opcion= int(input('Digite una opción (1-5): '))
    if opcion == 1 :
        usuarios = UsuarioDAO.seleccionar() #desde personaDAO se añade listar
        for usuario in usuarios:
            log.info(usuario)

    elif opcion == 2:
        username_var = input('Digite el nombre de usuario: ')
        password_var = input('Digite contraseña: ')
        usuario = Usuario(username=username_var, password=password_var)
        usuario_insertado = UsuarioDAO.insertar(usuario)
        log.info(f'Usuario insertado: {usuario_insertado}')

    elif opcion == 3:
        id_usuario_var = int(input('Digite id de usuario a modificar: '))
        username_var = input('Digite el nombre del usuario a modificar: ')
        password_var = input('Digite la contraseña del usuario a modificar: ')
        usuario = Usuario(id_usuario=id_usuario_var, username=username_var, password=password_var)
        usuario_actualizado = UsuarioDAO.actualizar(usuario)
        log.info(f'Usuario actualizado: {usuario_actualizado}')
    elif opcion == 4:
        id_usuario_var= int(input('Digite el usuario  a eliminar'))
        usuario = Usuario(id_usuario=id_usuario_var)
        usuario_eliminado= UsuarioDAO.eliminar(usuario)
        log.info(f'Usuario eliminado: {usuario_eliminado}')
else:
    log.info('Salimos de la Aplicación, Hasta pronto!')

#Fin Ejercicio Matías

#Ejercicios Jose Remaggi

from capa_datos_personas.Usuario import Usuario
from capa_datos_personas.usuario_dao import UsuarioDAO
from logger_base import log


opcion = None
while opcion != 5:
    print('Opciones: ')
    print('1.Listar Usuarios')
    print('2.Agregar Usuarios')
    print('3.Modificar usuarios')
    print('4.Eliminar usuario')
    print('5.Salir')
    opcion= int(input('Digite una opción (1-5): '))
    if opcion == 1 :
        usuarios = UsuarioDAO.seleccionar() #desde personaDAO se añade listar
        for usuario in usuarios:
            log.info(usuario)

    elif opcion == 2:
        username_var = input('Digite el nombre de usuario: ')
        password_var = input('Digite contraseña: ')
        usuario = Usuario(username=username_var, password=password_var)
        usuario_insertado = UsuarioDAO.insertar(usuario)
        log.info(f'Usuario insertado: {usuario_insertado}')

    elif opcion == 3:
        id_usuario_var = int(input('Digite id de usuario a modificar: '))
        username_var = input('Digite el nombre del usuario a modificar: ')
        password_var = input('Digite la contraseña del usuario a modificar: ')
        usuario = Usuario(id_usuario=id_usuario_var, username=username_var, password=password_var)
        usuario_actualizado = UsuarioDAO.actualizar(usuario)
        log.info(f'Usuario actualizado: {usuario_actualizado}')
    elif opcion == 4:
        id_usuario_var= int(input('Digite el usuario  a eliminar'))
        usuario = Usuario(id_usuario=id_usuario_var)
        usuario_eliminado= UsuarioDAO.eliminar(usuario)
        log.info(f'Usuario eliminado: {usuario_eliminado}')
else:
    log.info('Salimos de la Aplicación, Hasta pronto!')

#Fin Ejercicio Jose Remaggi

#Ejercicio Anabel Alesci
from logger_base import log
class UsuarioDAO:
    '''
    DAO -> Data Access Object para la tabla usuario
    CRUD -> Create-read-update-delete
    '''
    _SELECT = 'SELECT * FROM usuario ORDER BY id_usuario'
    _INSERTAR = 'INSERT INTO usuario(username, password) VALUES (%s,%s)'
    _ACTUALIZAR = 'UPDATE usuario SET username=%s, password=%s WHERE id_usuario=%s'
    _ELIMINAR = 'DELETE FROM usuario WHERE id_usuario=%s'

    @classmethod
    def seleccionar(cls):
        with CursorDelPool() as cursor:
            log.debug('Seleccionando usuarios')
            cursor.execute(cls._SELECT)
            registros = cursor.fetchall()
            usuarios = []
            for registro in registros:
                usuario = Usuario(registro[0], registro[1], registro[2])
                usuarios.append(usuario)
            return usuarios
    @classmethod
    def insertar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a insertar: {usuario}')
            valores = (usuario.username, usuario.password)
            cursor.execute(cls._INSERTAR, valores)
            return cursor.rowcount

    @classmethod
    def actualizar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a actualizar {usuario}')
            valores = (usuario.username, usuario.password, usuario.id_usuario)
            cursor.execute(cls._ACTUALIZAR, valores)
            return cursor.rowcount

    @classmethod
    def eliminar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a eliminar: {usuario}')
            valores=(usuario.id_usuario,)
            cursor.execute(cls._ELIMINAR,valores)
            return cursor.rowcount

#Fin ejercicio Anabel Alesci

#Ejercicio Yesica López

from capa_datos_persona.Usuario import Usuario
from capa_datos_persona.cursor_del_pool import CursorDelPool
from logger_base import log

class UsuarioDAO:
    '''
    DAO  ->Data Access Object Para la tabla de usuario
    CRUD ->Create - Read - Update - Delete
    '''

    _SELECT = 'SELECT * FROM usuario ORDER BY id_usuario'
    _INSERTAR = 'INSERT INTO usuario(username, password) VALUE (%s, %s)'
    _ACTUALIZAR = 'UPDATE usuario SET username=%s, password=%s WHERE id_usuario=%s'
    _ELIMINAR = 'DELETE FROM usuario WHERE id_usuario=%s'

    @classmethod
    def seleccionar(cls):
        with CursorDelPool() as cursor:
            log.debug('Seleccionando usuarios')
            cursor.execute(cls._SELECT)
            registros = cursor.fetchall()
            usuarios = []
            for registro in registros:
                usuario = Usuario(registro[0], registro[1], registro[2])
                usuarios.append(usuario)
            return usuarios

    @classmethod
    def insertar(cls,usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a insertar: {usuario}')
            valores = (usuario.username, usuario.password)
            cursor.execute(cls._INSERTAR, valores)
            return cursor.rowcount

    @classmethod
    def actualizar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a actualizar: {usuario}')
            valores = (usuario.username, usuario.password, usuario.id_usuario)
            cursor.execute(cls._ACTUALIZAR, valores)
            return cursor.rowcount

    @classmethod
    def eliminar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a eliminar: {usuario}')
            valores = (usuario.id_usuario,)
            cursor.execute(cls._ELIMINAR, valores)
            return cursor.rowcount

if __name__=='__main__':
    #Eliminar usuario
    usuario = Usuario(id_usuario=3)
    usuario_eliminado = UsuarioDAO.eliminar(usuario)
    log.debug(f'Usuario eliminado: {usuario_eliminado}')

    # Actualizar usuario
    usuario = Usuario(id_usuario=1, username='', password='369')
    usuario_modificado = UsuarioDAO.actualizar(usuario)
    log.debug(f'Usuario modificado: {usuario_modificado}')

    # Insertar usuario
    usuario = Usuario(username='kely', password='321')
    usuario_insertado = UsuarioDAO.insertar(usuario)
    log.debug(f'Usuario eliminado: {usuario_eliminado}')

    # Listar o seleccionar
    usuarios = UsuarioDAO.seleccionar()
    for usuario in usuarios:
    log.debug(usuario)

#Fin ejercicio Yesica López

# Ejercicio Jesús Mercado

from capa_datos_persona.Usuario import Usuario
from capa_datos_persona.cursor_del_pool import CursorDelPool
from logger_base import log

class UsuarioDAO:
    '''
    DAO -> Data Access Object Para la tabla de ususario
    CRUD -> Create - Read - Update - Delete
    '''

    _SELECT = 'SELECT * FROM usuario ORDER BY id_usuario'
    _INSERTAR = 'INSERT INTO usuario(username, password) VALUES (%s, %s)'
    _ACTUALIZAR = 'UPDATE usuario SET username=%s, password=%s WHERE id_usuario=%s'
    _ELIMINAR = 'DELETE FROM usuario WHERE id_usuario=%s'

    @classmethod
    def seleccionar(cls):
        with CursorDelPool() as cursor:
            log.debug('Seleccionar usuarios')
            cursor.execute(cls._SELECT)
            registros = cursor.fetchall()
            usuarios = []
            for registro in registros:
                usuario = Usuario(registro[0], registro[1], registro[2])
                usuarios.append(usuario)
            return usuarios

    @classmethod
    def insertar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a insertar: {usuario}')
            valores = (usuario.username, usuario.password)
            cursor.execute(cls._INSERTAR, valores)
            return cursor.rowcount

    @classmethod
    def actualizar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a actualizar: {usuario}')
            valores = (usuario.username, usuario.password, usuario.id_usuario)
            cursor.execute(cls._ACTUALIZAR, valores)
            return cursor.rowcount

    @classmethod
    def eliminar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a eliminar: {usuario}')
            valores = (usuario.id_usuario,)
            cursor.execute(cls._ELIMINAR, valores)
            return cursor.rowcount

if __name__ == '__main__':
    # Eliminar usuario
    usuario = Usuario(id_usuario=3)
    usuario_eliminado = UsuarioDAO.eliminar(usuario)
    log.debug(f'Usuario eliminado: {usuario_eliminado}')

    # Actualizar usuario
    usuario = Usuario(id_usuario=1, username='', password='369')
    usuario_modificado = UsuarioDAO.actualizar(usuario)
    log.debug(f'Usuario modificado: {usuario_modificado}')

    # Insertar usuario
    usuario = Usuario(username='Kely', password='321')
    usuario_insertado = UsuarioDAO.insertar(usuario)
    log.debug(f'Usuario insertado: {usuario_insertado}')

    # Listar usuario
    usuarios = UsuarioDAO.seleccionar()
    for usuario in usuarios:
        log.debug(usuario)


# Fin Ejercicio Jesús Mercado

#Ejercicio Victoria Zaccaro
from Usuario import Usuario
from cursor_del_pool import CursorDelPool
from logger_base import log

class UsuarioDAO:
    '''
    DAO -> Data Access Object Para la tabla de usuario
    CRUD -> Create - Read - Update - Delete
    '''

    _SELECT = 'SELECT * FROM usuario ORDER BY id_usuario'
    _INSERTAR = 'INSERT INTO usuario (username, password) VALUES (%s, %s)'
    _ACTUALIZAR = 'UPDATE usuario SET username=%s, password = %s WHERE id_usuario=%s'
    _ELIMINAR = 'DELETE FROM usuario WHERE id_usuario=%s'

    @classmethod
    def seleccionar(cls):
        with CursorDelPool() as cursor:
            log.debug('Seleccionando usuarios')
            cursor.execute(cls._SELECT)
            registros = cursor.fletchall()
            usuarios = []
            for registro in registros:
                usuario = Usuario(registro[0], registro[1], registro[2])
                usuarios.append(usuario)
            return usuarios

    @classmethod
    def insertar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a insertar: {usuario}')
            valores = (usuario.username, usuario.password)
            cursor.execute(cls._INSERTAR, valores)
            return cursor.rowcount

    @classmethod
    def actualizar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a actualizar: {usuario}')
            valores = (usuario.username, usuario.password, usuario.id_usuario)
            return cursor.rowcount

    @classmethod
    def eliminar(cls, usuario):
        with CursorDelPool() as cursor:
            log.debug(f'Usuario a eliminar: {usuario}')
            valores=(usuario.id_usuario,)
            cursor.execute(cls._ELIMINAR, valores)
            return cursor.rowcount

if __name__ == '__main__':
    # Eliminar Usuario
    usuario = Usuario(id_usuario=3)
    usuario_eliminado = UsuarioDAO.eliminar(usuario)
    log.debug(f'Usuario eliminado: {usuario_eliminado}')

    # Actualizar usuario
    usuario = Usuario(id_usuario=1, username='', password='369')
    usuario_modificado = UsuarioDAO.actualizar(usuario)
    log.debug(f'usuario modificado: {usuario_modificado}')

    # insertar usuario
    usuario = Usuario(username='kely', password='321')
    usuario_insertado = UsuarioDAO.insertar(usuario)
    log.debug(f'Usuario insertado: {usuario_insertado}')
    
    # Listar o seleccionar
    usuarios = UsuarioDAO.seleccionar()
    for usuario in usuarios:
        log.debug(usuario)
#Fin ejercicio Victoria Zaccaro
